backend/routes/post.py

```
@@ -254,19 +258,13 @@ async def create_event_post(
 
     return new_event  # Returns as EventResponse schema
 
-@router.get("/get_post/{post_id}")
-def get_post(post_id: int, current_user: User = Depends(get_current_user), db: Session = Depends(get_db)):
-    post = db.query(Post).filter(Post.id == post_id, Post.user_id == current_user.id).first()
-    if not post:
-        raise HTTPException(status_code=404, detail=STATUS_404_ERROR)
+@router.get("/posts/")
+def get_posts(user_id: Optional[int] = None, db: Session = Depends(get_db)):
```

Ruff (0.8.2)

Missing return type annotation for public function `get_posts`

(ANN201)